File: omnia_client/lib/models/offer.dart
class Offer {
  final String id;
  final String category;
  final String description;
  String? pictureUrl;
  final String title;

  Offer({
    required this.id,
    required this.category,
    required this.description,
    this.pictureUrl,
    required this.title,
  });

  Map<String, dynamic> toJson() {
    return {
      "category": category,
      "description": description,
      "picture_url": pictureUrl,
      "title": title,
      "id": id
    };
  }
  factory Offer.fromJson(Map<String,dynamic> json){
    return Offer(
      category: json['category'],
      description: json['description'],
      id: json['id'],
      pictureUrl: json['picture_url'],
      title: json['title']
    );
  }
}

File: admin/app/src/main/java/com/yousef_shora/omniaoffers/pojo/Offer.java
package com.yousef_shora.omniaoffers.pojo;

import com.yousef_shora.omniaoffers.modal.Utils;

import java.io.Serializable;
import java.util.UUID;

public class Offer  implements Serializable {
    public String title;
    public String description = "";
    public String picture_url;
    public String category;
    public final String id;

    public Offer(String title, String picture_url, String category) {
        this.title = title;
        if (Utils.is_category(category)) {
            this.category = category;
        }
        this.picture_url = picture_url;
        id = UUID.randomUUID().toString();
    }

    public Offer(String title, String picture_url, String category, String description) {
        this.title = title;
        if (Utils.is_category(category)) {
            this.category = category;
        }
        this.picture_url = picture_url;
        this.description = description;
        id = UUID.randomUUID().toString();
    }

    public Offer(String title, String category) {
        this.title = title;
        if (Utils.is_category(category)) {
            this.category = category;
        }
        id = UUID.randomUUID().toString();
    }

    public Offer() {
        id = UUID.randomUUID().toString();

    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        if (Utils.is_category(category)) {
            this.category = category;
        }
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getPicture_url() {
        return picture_url;
    }

    public void setPicture_url(String picture_url) {
        this.picture_url = picture_url;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getId() {
        return id;
    }
}